demo/lib/distro.bash

```diff
@@ -0,0 +1,322 @@
+        debian-refresh-pkg-db
+        debian-install-pkg containerd
+        generic-setup-containerd
+    }
+}
+
+debian-install-containerd() {
+    vm-command-q "[ -f /usr/bin/containerd ]" || {
+        debian-refresh-pkg-db
+        debian-install-pkg containerd
+        generic-setup-containerd
+    }
+}
+
+debian-install-k8s() {
+    debian-refresh-pkg-db
+    debian-install-pkg apt-transport-https curl
+    debian-install-repo-key "https://packages.cloud.google.com/apt/doc/apt-key.gpg"
+    debian-install-repo "deb https://apt.kubernetes.io/ kubernetes-xenial main"
+    debian-install-pkg kubeadm kubelet kubectl
+    vm-command "apt-get update && apt-get install -y kubelet kubeadm kubectl" ||
+        command-error "failed to install kubernetes packages"
+}
+
+debian-set-kernel-cmdline() {
+    local e2e_defaults="$@"
+    vm-command "echo 'GRUB_CMDLINE_LINUX_DEFAULT=\"\${GRUB_CMDLINE_LINUX_DEFAULT} ${e2e_defaults}\"' > /etc/default/grub.d/60-e2e-defaults.cfg" || {
+        command-error "writing new command line parameters failed"
+    }
+    vm-command "update-grub" || {
+        command-error "updating grub failed"
+    }
+}
+
+
+###########################################################################
+
+#
+# Centos 7, 8, generic Fedora, generic rpm functions
+#
+
+YUM_INSTALL="yum install --disableplugin=fastestmirror -y"
+
+centos-7-image-url() {
+    echo ="https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1503.qcow2.xz"
+}
+
+centos-8-image-url() {
+    echo "https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.2.2004-20200611.2.x86_64.qcow2"
+}
+
+centos-ssh-user() {
+    echo centos
+}
+
+centos-7-install-repo() {
+    vm-command-q "$type -t yum-config-manager >&/dev/null" || {
+        distro-install-pkg yum-utils
+    }
+    vm-command "yum-config-manager --add-repo $*" ||
+        command-error "failed to add YUM repository $*"
+}
+
+centos-7-install-pkg() {
+    vm-command "$YUM_INSTALL $*" ||
+        command-error "failed to install $*"
+}
+
+centos-install-golang() {
+    distro-install-pkg wget tar gzip
+    from-tarball-install-golang
+}
+
+fedora-image-url() {
+    echo "https://download.fedoraproject.org/pub/fedora/linux/releases/32/Cloud/x86_64/images/Fedora-Cloud-Base-32-1.6.x86_64.qcow2"
+}
+
+fedora-ssh-user() {
+    echo fedora
+}
+
+fedora-install-repo() {
+    distro-install-pkg dnf-plugins-core
+    vm-command "dnf config-manager --add-repo $*" ||
+        command-error "failed to install DNF repository $*"
+}
+
+fedora-install-pkg() {
+    vm-command "dnf install -y $*" ||
+        command-error "failed to install $*"
+}
+
+fedora-install-golang() {
+    fedora-install-pkg wget tar gzip
+    from-tarball-install-golang
+}
+
+fedora-install-containerd() {
+    vm-command-q "[ -f /usr/bin/containerd ]" || {
+        distro-install-repo https://download.docker.com/linux/centos/docker-ce.repo
```

Comment:
for fedora, exists https://download.docker.com/linux/fedora/

Also, should we probably use containerd that shipped in distro?

Reply:
Ok, I'll update and see if the shipped one works.